### PR TITLE
Bridging between L1 ETH and L2 ERC20 BridgedETH

### LINKED ISSUE #63: Bridge ETH to L2
The current wrapped ETH tokens on Celo are bridged through Axelar and Wormhole (two separate tokens). If we add native bridging, we would get a new ETH token on CELO (and use unwrapped ETH, if possible).

The default token contract for bridged tokens on the L2 is https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol. If we add `debitGasFee` and `creditGasFee`, we could use bridged tokens as gas currencies (see https://github.com/celo-org/celo-blockchain-planning/issues/902).

See
* [L1StandardBridge](https://github.com/ethereum-optimism/optimism/blob/beeffe9cca7cbd2c8d9b82fe7bbdb292c3e103db/packages/contracts-bedrock/src/L1/L1StandardBridge.sol)
* [L2StandardBridge](https://github.com/ethereum-optimism/optimism/blob/beeffe9cca7cbd2c8d9b82fe7bbdb292c3e103db/packages/contracts-bedrock/src/L2/L2StandardBridge.sol)
* [Bridging tutorial](https://github.com/ethereum-optimism/optimism-tutorial/tree/main/standard-bridge-standard-token), [bridging tutorial (custom token)](https://github.com/ethereum-optimism/optimism-tutorial/tree/main/standard-bridge-custom-token)
```[tasklist]
### Tasks
- [x] Deploy ERC20 token for ETH on L2
- [x] Bridge L1 ETH to an ERC20 token on L2
- [x] Bridge ERC20 ETH from L1 to L1 ETH
- [x] Add test
- [ ] Adapt existing optimism bridging tests
```

### PR DESCRIPTION
**Pull request by karlb**
* Add a basic ERC20 contract on L2 which is used to stored bridged ETH. This can be extended to be a fee currency later.
* Configure bridging between L1 ETH and L2 bridged ETH
* Add helper code to prove and finalize L2->L1 withdrawals in tests
* Test bridging in both directions

Notes:
* Before Bedrock, Optimism also used an ERC20 token to store ETH on L2. But since a lot changed during Bedrock, I didn't try to revive the old version and modified the new one, instead.
* It is not totally clear whether the ETH should be held in the L1StandardBridge contract or OptimisimPortal on L1. I went with the former, since the it seems to be the expectation that all ETH send to OptimisimPortal should come out as native currency on L2.
* Any good suggestions how to make sure I didn't miss a way to trigger the old native<->native bridging?

Closes https://github.com/celo-org/optimism/issues/63

### CODE REVIEW

File: packages/contracts-bedrock/src/L1/L1StandardBridge.sol [hunk 1]
```diff
@@ -211,7 +211,9 @@ contract L1StandardBridge is StandardBridge, ISemver {
     /// @param _minGasLimit Minimum gas limit for the deposit message on L2.
     /// @param _extraData   Optional data to forward to L2.
     function _initiateETHDeposit(address _from, address _to, uint32 _minGasLimit, bytes memory _extraData) internal {
-        _initiateBridgeETH(_from, _to, msg.value, _minGasLimit, _extraData);
+        _initiateBridgeETHToERC20(
+            0x4200000000000000000000000000000000001023, _from, _to, msg.value, _minGasLimit, _extraData
```
Comment (gastonponti) [inline packages/contracts-bedrock/src/L1/L1StandardBridge.sol]: Maybe set this as a constant with a name? But I don't totally know if that could cost or not more gas to be honest

File: packages/contracts-bedrock/src/L2/L2StandardBridge.sol [hunk 2]
```diff
@@ -175,7 +173,9 @@ contract L2StandardBridge is StandardBridge, ISemver {
     )
         internal
     {
-        if (_l2Token == Predeploys.LEGACY_ERC20_ETH) {
+        if (_l2Token == address(0x4200000000000000000000000000000000001023)) {
```
Comment (gastonponti) [inline packages/contracts-bedrock/src/L2/L2StandardBridge.sol]: Same

File: packages/contracts-bedrock/src/libraries/Predeploys.sol [hunk 3]
```diff
@@ -74,4 +74,7 @@ library Predeploys {
 
     /// @notice Address of the EAS predeploy.
     address internal constant EAS = 0x4200000000000000000000000000000000000021;
+
+    /// @notice Address of the BridgedETH predeploy.
+    address internal constant BRIDGED_ETH = 0x4200000000000000000000000000000000001023;
```
Comment (gastonponti) [inline packages/contracts-bedrock/src/libraries/Predeploys.sol:79]: Well, it's already set here

File: packages/contracts-bedrock/src/universal/StandardBridge.sol [hunk 4]
```diff
@@ -152,7 +152,9 @@ abstract contract StandardBridge is Initializable {
     ///                     not be triggered with this data, but it will be emitted and can be used
     ///                     to identify the transaction.
     function bridgeETH(uint32 _minGasLimit, bytes calldata _extraData) public payable onlyEOA {
-        _initiateBridgeETH(msg.sender, msg.sender, msg.value, _minGasLimit, _extraData);
+        _initiateBridgeETHToERC20(
+            0x4200000000000000000000000000000000001023, msg.sender, msg.sender, msg.value, _minGasLimit, _extraData
```
Comment (gastonponti) [inline packages/contracts-bedrock/src/universal/StandardBridge.sol]: same

File: packages/contracts-bedrock/src/universal/StandardBridge.sol [hunk 5]
```diff
@@ -168,7 +170,9 @@ abstract contract StandardBridge is Initializable {
     ///                     not be triggered with this data, but it will be emitted and can be used
     ///                     to identify the transaction.
     function bridgeETHTo(address _to, uint32 _minGasLimit, bytes calldata _extraData) public payable {
-        _initiateBridgeETH(msg.sender, _to, msg.value, _minGasLimit, _extraData);
+        _initiateBridgeETHToERC20(
+            0x4200000000000000000000000000000000001023, msg.sender, _to, msg.value, _minGasLimit, _extraData
```
Comment (gastonponti) [inline packages/contracts-bedrock/src/universal/StandardBridge.sol]: same

**Review by gastonponti — COMMENTED** [review]
Looks good. Just comments for Constants usage

File: packages/contracts-bedrock/src/L1/L1StandardBridge.sol (hunk 1 above)
Comment (karlb) [reply packages/contracts-bedrock/src/L1/L1StandardBridge.sol]: Ahh yes. When I started out I was not sure were to put the named constant, and later I just forgot. Fixed now.

**Comment by karlb** [comment]
There's on test waiting to be updated, but I'll do that in a second PR if this one get's ready to merge quickly.

**Review by carterqw2 — APPROVED** [review]

**Pull request merged**